### PR TITLE
Refactor/vue js implementation

### PR DESCRIPTION
**Pull request by vallo87**
first basic vue implementation in be4-web

- the usual Vue templating delimiters ( {{}} ) were changed to @() due to collision with twig delimiters;
- scripts are simply loaded in default layout for now;
- first refactor of modules index template;
- js app folder structured under webroot/js/app:
  -  app.js
  -  modules/index.js (modules is the type of page folder (module, login, etc), index.js is related to the index method in bedita AppController, as so view.js): Vue component for the index of the module
  -  modules/view.js: Vue component for the view of the module
- index.js defines a vue component <modules-index> that wraps what's inside Template/Modules/index.twig;
- view.js defines a vue component <modules-index> that wraps what's inside Template/Modules/view.twig (view was not yet refactored);
- to be able to include pagination toolbar in index vue component a little refactor was made: the toolbar was moved inside index.twig, from default.twig;
- basic vue configuration was placed under webroot/js/config/config.js

... needless to say more needs to be done!

### CODE REVIEW

**Comment by stefanorosanelli** [comment]
please follow our https://github.com/bedita/bedita/wiki/Commit-conventions 
so, you may have:
* `[minor] chore: ....`
* `chore: ...` or `feat: ...`

**Comment by stefanorosanelli** [comment]
Seems ok to me, for what I can understand anyway :wink: ...
Please @didoda @batopa share your thoughts

File: webroot/js/app/modules/index.js [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+     * component methods
+     */
+    methods: {
+
+        /**
+         * extract params from page url
+         *
+         * @returns {void}
+         */
+        loadUrlParams() {
```
Comment (didoda) [inline webroot/js/app/modules/index.js]: `loadUrlParams` e  `buildUrlParams` rendono deprecati/inutili i metodi `LinkHelper->page`, `LinkHelper->pageSize` e `LinkHelper->query`? In tal caso valutiamo se eliminarli. Io avrei preferito demandare all'helper la gestione dei link, ma non so se usando `vue` ciò è possibile.

Comment (stefanorosanelli) [reply webroot/js/app/modules/index.js]: yes, è vero, non avevo visto, meglio discuterne

**Pull request merged**